### PR TITLE
feat(Kandel): Add getMissingProvisionFromOffers

### PR DESCRIPTION
**Pull request by lnist**
* chore(OfferLogic): Refactor getMissingProvision for offerLogic and liquidityProvider
* fix(Kandel): make recommended config just default
* feat(Kandel): Add getMissingProvisionFromOffers
* fix(Kandel): Update kandelConfiguration based on latest knowledge

Also fixes https://github.com/mangrovedao/mangrove-ts/issues/557

### CODE REVIEW

File: packages/mangrove.js/src/kandel/kandelInstance.ts [hunk 1]
```diff
@@ -685,13 +685,98 @@ class KandelInstance {
...
+    gasreq?: number;
+    gasprice?: number;
+    distribution?: KandelDistribution;
+    offerCount?: number;
+  }) {
+    const existingOffers = (await this.getOffers()).map((x) => ({
+      gasprice: x.offer.gasprice,
+      gasreq: x.offer.gasreq,
+      gasbase: x.offer.offer_gasbase,
+    }));
```
Comment (peterMangrove) [inline packages/mangrove.js/src/kandel/kandelInstance.ts:732]: Consider moving this to its own method, as you are doing this in at least 2 places.

**Review by peterMangrove — CHANGES_REQUESTED** [review]
lgtm. Only a few comments

File: packages/mangrove.js/src/kandel/kandelInstance.ts [hunk 2]
```diff
@@ -685,13 +685,98 @@ class KandelInstance {
     );
   }
 
+  /** Calculates the provision locked by existing offers based on the given parameters
+   * @returns the locked provision, in ethers.
```
[the remark this answers is not in the document]
Comment (lnist) [reply packages/mangrove.js/src/kandel/kandelInstance.ts:689]: yes, it's what we use in multiple places.

File: packages/mangrove.js/src/kandel/kandelInstance.ts [hunk 3]
```diff
@@ -685,13 +685,98 @@ class KandelInstance {
...
+   */
+  public async getLockedProvision() {
+    const existingOffers = (await this.getOffers()).map((x) => ({
+      gasprice: x.offer.gasprice,
+      gasreq: x.offer.gasreq,
+      gasbase: x.offer.offer_gasbase,
+    }));
+    return this.market.mgv.calculateOffersProvision(existingOffers);
```
[the remark this answers is not in the document]
Comment (lnist) [reply packages/mangrove.js/src/kandel/kandelInstance.ts:697]: Good point, fixing.

File: packages/mangrove.js/src/kandel/kandelInstance.ts [hunk 4]
```diff
@@ -685,13 +685,98 @@ class KandelInstance {
...
+   * @param existingOffers[].gasbase the offer list's offer_gasbase.
+   * @returns the locked provision, in ethers.
+   */
+  public getLockedProvisionFromOffers(
+    existingOffers: { gasprice: number; gasreq: number; gasbase: number }[]
+  ) {
+    return this.market.mgv.calculateOffersProvision(existingOffers);
+  }
+
+  /** Gets the missing provision based on provision already available on Mangrove, potentially locked by existing offers. It assumes all locked provision will be made available via deprovision of due to offers being replaced.
```
[the remark this answers is not in the document]
Comment (lnist) [reply packages/mangrove.js/src/kandel/kandelInstance.ts:713]: Yse :)

File: packages/mangrove.js/src/kandel/kandelInstance.ts (hunk 1 above)
Comment (lnist) [reply packages/mangrove.js/src/kandel/kandelInstance.ts]: done

**Review by peterMangrove — APPROVED** [review]

**Pull request merged**